Fixes #370: Deprecation warnings on Django 3.x

jannh opened the pull request:
Since Django 1.11, which is the last version not supporting the new path syntax is not supported anymore, there should be no blockers merging this.
Btw: I found some import try/except blocks for python2/3 - if python2 is unsupported, should those be removed?

akatsoulas commented:
> Btw: I found some import try/except blocks for python2/3 - if python2 is unsupported, should those be removed?
Sounds good to me :+1: 

@jannh Tests are failing b/c there are references of Django 1.11 that should be removed

jannh commented:
I now fixed the test config (only Django >= 2.2 and corresponding supported python versions), fixed deprecation warnings and removed the python2 import shims.

akatsoulas reviewed (commented):
Thanks @jannh for this PR! I did a first pass and this looks good to me. I will have another more thorough one before merging. I won't merge it yet b/c this has backwards incompatible changes. I will do another release of the lib first and then bump the major version with this PR.

jannh commented:
Any news on this PR - it's been around for a while now :)

sbor23 commented on .travis.yml:
```diff
@@ -5,8 +5,6 @@ sudo: false
 python:
-  - "2.7"
-  - "3.4"
   - "3.5"
```
3.5 is EOL by now.

jannh replied:
Updated PR: Removed 3.5 and added 3.9

sbor23 commented on tox.ini:
```diff
@@ -1,20 +1,16 @@
-    py{34,35,36,37}-django200
-    py{35,36,37}-django210
-    py{35,36,37}-django220
+    py{35,36,37,38}-django220
```
drop 35, add 39

jannh replied:
Updated PR: Removed 3.5 and added 3.9

sbor23 commented on tox.ini:
```diff
@@ -1,20 +1,16 @@
-    py{35,36,37}-django220
+    py{35,36,37,38}-django220
     py{36,37,38}-django300
+    py{36,37,38}-django310
```
add 39

jannh replied:
Updated PR: Removed 3.5 and added 3.9

sbor23 commented on tox.ini:
```diff
@@ -1,20 +1,16 @@
-  3.7: py37, coverage, lint
-  3.8: py38
+  3.7: py37
+  3.8: py38, coverage, lint
```
same here

jannh replied:
Updated PR: Removed 3.5 and added 3.9

sbor23 reviewed (commented):
Hey, some remarks since this PR was lying around for long...
Needs a rebase on master too, there are some conflicts.

I really hope this finally gets some attention by the maintainers.

jannh commented:
Removed python 3.5, added python 3.9 to tests.
Rebased to current master branch

johngian commented on tox.ini:
```diff
@@ -1,20 +1,16 @@
-  3.4: py34
-  3.5: py35
   3.6: py36
-  3.7: py37, coverage, lint
```
This should be adapted in the codecov gh action. But i don't think it should block this PR.

akatsoulas reviewed (approved):
Thanks @jannh for this PR. :rocket:

akatsoulas commented:
Closing in favor of #403 which removes travis
